src/linear_algebra/multilinear/finite_dimensional.lean

```diff
@@ -4,38 +4,59 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Oliver Nash
 -/
 import linear_algebra.multilinear.basic
-import linear_algebra.matrix.to_lin
+import linear_algebra.free_module.finite.basic
 
 /-! # Multilinear maps over finite dimensional spaces
 
-The main result of this file is `multilinear_map.finite_dimensional`.
+The main results are that multilinear maps over modules that are both free and finite are
+themselves both free and finite as modules.
 
-We do not put this in `linear_algebra/multilinear_map/basic`, as `matrix.to_lin` pulls in a lot of
-imports, which can cause timeouts downstream.
+* `module.finite.multilinear_map`
+* `module.free.multilinear_map`
+
+We do not put this in `linear_algebra/multilinear_map/basic` to avoid making the imports too large
+there.
 -/
 
 namespace multilinear_map
 
 variables {ι R M₂ : Type*} {M₁ : ι → Type*}
 variables [decidable_eq ι]
-variables [fintype ι] [field R] [add_comm_group M₂] [module R M₂] [finite_dimensional R M₂]
-variables [∀ i, add_comm_group (M₁ i)] [∀ i, module R (M₁ i)] [∀ i, finite_dimensional R (M₁ i)]
+variables [fintype ι] [comm_ring R] [add_comm_group M₂] [module R M₂]
+variables [Π i, add_comm_group (M₁ i)] [Π i, module R (M₁ i)]
+variables [module.finite R M₂] [module.free R M₂]
+variables [∀ i, module.finite R (M₁ i)] [∀ i, module.free R (M₁ i)]
 
-instance : finite_dimensional R (multilinear_map R M₁ M₂) :=
+-- the induction requires us to show both at once
```

Comment:
I doubt you intended this but this wording suggests to me that the conclusion fails with just one property (which is not true).
```suggestion
```

Reply:
Sorry, I don't understand what you're getting at; my claim is that the conclusion _does_ fail with just one property, and that both halves of the condition need both results from the previous inductive step.

Are we splitting hairs over wording, or is there a mathematical generalization that's been missed here?

Reply:
We were not splitting hairs.

However having started to sketch out what I had in mind I realise it was not correct. I retract this remark. Sorry for wasting time!

Reply:
I also thought briefly about the generalization, but all the lemmas that I wanted to use didn't have it so I gave up on caring whether it was actually true!